components/infobox/wikis/stormgate/infobox_building_custom.lua

```diff
@@ -194,19 +200,35 @@ function CustomBuilding:setLpdbData(args)
 	})
 end
 
----@return string[]
-function CustomBuilding:_getArmorDisplay()
-	local armorTypes = self:_displayCommaSeparatedString(self.args.armor_type)
+---@return string?
+function CustomBuilding:_getDefenseDisplay()
+	local args = self.args
+	local health = tonumber(args.health)
+	local extraHealth = health and tonumber(args.extra_health)
+	local armor = tonumber(args.armor)
+
+	return table.concat(Array.append({},
+		health and ICON_HP or nil,
+		health,
+		extraHealth and ('(+' .. extraHealth .. ')') or nil,
+		armor and ICON_ARMOR or nil,
+		armor
+	), '&nbsp;')
+end
 
-	return Array.append({},
-		self.args.armor and (ICON_ARMOR .. ' ' .. self.args.armor) or nil,
-		String.nilIfEmpty(table.concat(armorTypes, ', '))
-	)
+---@param inputString string?
+---@return string
+function CustomBuilding:_displayCommaSeparatedString(inputString)
+	return table.concat(Array.map(Array.parseCommaSeparatedString(inputString),
+		function(value)
+			return Page.makeInternalLink(value)
+		end
+	), ', ')
 end
 
 ---@param inputString string?
 ---@return string[]
-function CustomBuilding:_displayCommaSeparatedString(inputString)
+function CustomBuilding:_displayCommaSeparatedStringWithBreaks(inputString)
```

Comment:
heavily dislike basically doubling a function
as mentioned before:
- maybe a `._csvToPageList` function instead of the csv display functions
- could then add `function ._displayArrayAsCsv(input) return table.concat(._csvToPageList(input), ', ') end`